source/common/http/async_client_impl.cc

```diff
@@ -186,34 +188,27 @@ void AsyncRequestImpl::initialize() {
   sendHeaders(request_->headers(), !request_->body());
   if (!remoteClosed() && request_->body()) {
     sendData(*request_->body(), true);
+  } else if (remoteClosed() && request_->body()) {
+    closeLocal(true);
```

Comment:
logic can be simplified by first wrapping with
```
if (request_->body()) {
  if (!remoteClosed() {...}
  else {...}
}
```

Reply:
This seems to really only protect against the case where the remote is closed synchronously during the call to `sendHeaders`. Is that sufficient to handle edge cases?